wrappers/rust/icicle-core/src/vec_ops/mod.rs

```diff
@@ -207,6 +213,23 @@ where
     <<F as FieldImpl>::Config as VecOps<F>>::add(a, b, result, &cfg)
 }
 
+pub fn accumulate_scalars<F>(
+    a: &mut (impl HostOrDeviceSlice<F> + ?Sized),
+    b: &(impl HostOrDeviceSlice<F> + ?Sized),
+    cfg: &VecOpsConfig,
+) -> IcicleResult<()>
+where
+    F: FieldImpl,
+    <F as FieldImpl>::Config: VecOps<F>,
+{
+    assert_eq!(
+        cfg.is_a_on_device, cfg.is_result_on_device,
+        "input vector 'a' must have same 'on device' setting as result, since operation is performed in-place"
+    );
```

Comment:
Is this needed? These values are not publicly accessible and are updated in the next line via `check_vec_ops_args` where `a` is passed as both `a` and `result` so it will always be true

Reply:
removed